code/src/containers/dashboard-tabs/components/orders-tab-filter-drawer/OrdersTabFilterDrawer.tsx

```diff
@@ -1,36 +1,137 @@
+import { DeliveryMethod } from "@/types/delivery.types";
+import { OrderStatus } from "@/types/order.types";
 
 import "@/containers/dashboard-tabs/components/orders-tab-filter-drawer/OrdersTabFilterDrawer.scss";
 
+import { AdminOrderFilters } from "../../hooks/use-filtered-admin-orders/useFilteredAdminOrders.types";
+
 type OrdersTabFilterDrawerProps = {
-  filtersTitleTranslationProps: TranslationProps;
+  activeFiltersCount: number;
+  filters: AdminOrderFilters;
+  filterActions: FilterActions<AdminOrderFilters>;
+  closeFilterDrawer: () => void;
 };
 
+type FiltersWithSets = Pick<AdminOrderFilters, "delivery-methods" | "statuses">;
+
 const OrdersTabFilterDrawer = ({
-  filtersTitleTranslationProps
+  activeFiltersCount,
+  filters,
+  filterActions,
+  closeFilterDrawer
 }: OrdersTabFilterDrawerProps) => {
-  const { closeDrawer } = useDrawerContext();
+  const selectLabelId = useId();
 
-  const handleApplyFilters = () => {
-    // @TODO: add logic of applying filters
-    closeDrawer();
+  const {
+    applyFilters,
+    checkFilterActive,
+    resetFilterByKey,
+    updateFilterByKey,
+    resetFilters
+  } = filterActions;
+
+  const handleCheckboxListChange =
+    <Key extends keyof FiltersWithSets>(
+      key: Key,
+      value: ExtractSetValue<FiltersWithSets[Key]>
+    ) =>
+    (event: SyntheticEvent, checked: boolean) => {
+      const filtersSet = new Set(filters[key]);
+
+      if (checked) {
+        filtersSet.add(value);
+      } else {
+        filtersSet.delete(value);
+      }
+
+      updateFilterByKey(key, filtersSet as AdminOrderFilters[Key]);
+    };
+
+  const orderStatusesCheckboxes = Object.entries(
+    orderStatusesTranslationKeys
+  ).map(([status, translationKey]) => (
+    <AppCheckbox
+      key={translationKey}
+      checked={filters.statuses.has(status as OrderStatus)}
+      onChange={handleCheckboxListChange("statuses", status as OrderStatus)}
+      variant="dark"
+      labelTranslationKey={translationKey}
+    />
+  ));
+
+  const isOrdersStatusFilterActive = checkFilterActive("statuses");
+
+  const handleOrderStatusFilterReset = () => {
+    resetFilterByKey("statuses");
+  };
+
+  const handleTimespanSelectChange = (event: SelectChangeEvent<unknown>) => {
+    updateFilterByKey("timespan", event.target.value as TimeSpan);
+  };
+
+  const timespanSelect = (
+    <AppSelect
+      inputProps={{
+        className: "order-tab-filters__date-period-select"
+      }}
+      displayEmpty
+      defaultValue=""
+      value={filters.timespan}
+      onChange={handleTimespanSelectChange}
+      labelId={selectLabelId}
+    >
+      <AppMenuItem value="" disabled>
+        <AppTypography translationKey="select.defaultOption" />
+      </AppMenuItem>
+      {Object.entries(timeSpans).map(([datePeriod, translationKey]) => (
+        <AppMenuItem value={datePeriod} key={datePeriod}>
+          <AppTypography translationKey={translationKey} />
+        </AppMenuItem>
+      ))}
```

Comment:
I think it would be better to move it above the return statement

Reply:
Thanks for advice! I did it!